song-server/src/main/java/bio/overture/song/server/service/StorageService.java

```diff
@@ -58,18 +58,19 @@
   @NonNull private final RetryTemplate retryTemplate;
   @NonNull private final String storageUrl;
   @NonNull private final ValidationService validationService;
+  @NonNull private final String scoreAuthorizationHeader;
 
   @SneakyThrows
   public boolean isObjectExist(@NonNull String accessToken, @NonNull String objectId) {
     return doGetBoolean(accessToken, getObjectExistsUrl(objectId));
   }
 
   @SneakyThrows
-  public StorageObject downloadObject(@NonNull String accessToken, @NonNull String objectId){
-    val objectExists = isObjectExist(accessToken, objectId);
+  public StorageObject downloadObject(@NonNull String objectId){
+    val objectExists = isObjectExist(scoreAuthorizationHeader, objectId);
```

Comment:
make `isObjectExist` internally depend on the member variable `scoreAuthorizationHeader`. After some discussion, the song user's authorization token has nothing to do with songs communication with the storage server. Therefore, all communication between the song-server and the score-server should be done with an application level access token with download scope (for downloading object specifications) and upload scope (for checking object existence). The song users authorization is handled by the song-server and must be specified in the song users authorization header of their request. Song handles the responsibility of authorizing users against its endpoints, when the storage service is used "internally" by song, which should have full access, for validating the users request data